src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -264,7 +264,12 @@ private static String getSkippedMessage(Element testCase) {
     }
 
     public String getDisplayName() {
-        return TestNameTransformer.getTransformedName(testName);
+        String archiveId = this.parent != null ? Util.fixEmptyAndTrim(parent.getArchiveId()) : null;
```

Comment:
calling` fixEmptyAndTrim` repeatedly is not the best as the String will be searched for spaces repeatedly.
better to `fixEmptyAndTrim` when it is set in the parent and just then call `parent.getArchiveId()`